controller/watch/watch_test.go

```diff
@@ -14,12 +14,125 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package watch_test
+package watch
 
 import (
+	"reflect"
 	"testing"
+	"time"
+
+	"k8s.io/client-go/1.5/pkg/api/unversioned"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	extv1beta1 "k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/1.5/pkg/watch"
 )
 
+func TestDeploymentWatcher(t *testing.T) {
+	const (
+		evtChTimeout = 100 * time.Millisecond
+	)
+	fakeWatcher := watch.NewFake()
+	deplIface := &fakeDeploymentInterface{
+		listRet: &extv1beta1.DeploymentList{
+			TypeMeta: unversioned.TypeMeta{},
+			ListMeta: unversioned.ListMeta{},
+			Items: []extv1beta1.Deployment{
+				extv1beta1.Deployment{
+					ObjectMeta: v1.ObjectMeta{Name: "listdepl1"},
+				},
+				extv1beta1.Deployment{
+					ObjectMeta: v1.ObjectMeta{Name: "listdepl2"},
+				},
+			},
+		},
+		watchRet: fakeWatcher,
+	}
+	evtCh := make(chan watch.Event)
+	wcb := func(evt watch.Event) error {
+		evtCh <- evt
+		return nil
+	}
+	go deploymentWatcher(deplIface, wcb)
+
+	// ensure list of deployments is processed first
+	for i, item := range deplIface.listRet.Items {
```

Comment:
@martinmaly yes. more precisely, this for loop verifies that the first events received in the channel are the ones returned from the deployments interface's `List()` func, and within that return value, they're the same order as expected